### DISCUSSION TITLE
Dehydrate query.promise called when there is no network connection

### DISCUSSION THREAD

**Discussion by ugurcanatas** [TanStack/query#8228] [Q&A]
Hey there,

I recently realised after upgrading to the latest react query version that there is an error from a certain query that we receive when user is offline.

Specifically:
```ts
console.error(
            `A query that was dehydrated as pending ended up rejecting. [${query.queryHash}]: ${error}; The error will be redacted in production builds`,
          )
```

Case steps:
- User doesn't have a network connection
- User opens the app
- Query options have `networkMode: 'online'` 
- Query is initially in a pending state.
- Query resolves to error state but user has no network connection with `Promise.reject(new Error('redacted'))`.

As far as I understand, when query is in the pending state, we are trying to resolve that pending promise.

My question, is this behaviour intended if app does not have any network connection?

**Comment by TkDodo** [comment]
how does the app receive the promise from the server if it doesn’t have network connection ?

**Reply by ugurcanatas** [reply]
That makes sense, maybe i phrased it wrong. It might be also related to the place when I set the onlineManager state. I'll try to create a repro and get back. Thanks
